addons/http_routing/models/ir_http.py

```diff
@@ -164,10 +165,46 @@ def url_for(path_or_uri, lang_code=None):
             elif lang_url_code != default_lg.url_code or force_lang:
                 ps.insert(1, lang_url_code)
             location = u'/'.join(ps)
-
     return location
 
 
+def url_for(url_from, lang_code=None, no_rewrite=False):
+    ''' Return the url with the rewriting applied.
+        Nothing will be done for absolute URL, or short URL from 1 char.
+
+        :param url_from: The URL to convert.
+        :param lang_code: Must be the lang `code`. It could also be something
+                          else, such as `'[lang]'` (used for url_return).
+        :param no_rewrite: don't try to match route with website.rewrite.
+    '''
+    new_url = False
+
+    # don't try to match route if we know that no rewrite has been loaded.
+    if not request.env['ir.http']._rewrite_len.get(request.website_routing):
```

Comment:
@JKE-be When website isn't installed, request.website_routing isn't initialized and this crashes.
Could you fix it? (If I do it, I'm just gonna c/p 2 lines of code from website to here :P)
https://github.com/odoo/odoo/issues/38020

`website/models/ir_http.py
`![image](https://user-images.githubusercontent.com/19364378/66331233-57ab6e00-e932-11e9-9cb3-7628af6c072e.png)